src/Acuminator/Acuminator.Analyzers/Analyzers/Localization/LocalizationInvocationAnalyzer.cs

```
@@ -0,0 +1,110 @@
+        private InvocationExpressionSyntax _invocationNode;
+        private bool _isFormatMethod;
+        private ExpressionSyntax _messageExpression;
+
+        private SemanticModel SemanticModel => _syntaxContext.SemanticModel;
+        private CancellationToken Cancellation => _syntaxContext.CancellationToken;
+        private LocalizationTypes Localization => _pxContext.Localization;
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            ImmutableArray.Create
+            (
+                Descriptors.PX1050_HardcodedStringInLocalizationMethod,
+                Descriptors.PX1051_NonLocalizableString,
+                Descriptors.PX1052_IncorrectStringToFormat,
+                Descriptors.PX1053_ConcatinationPriorLocalization
+            );
+
+        internal override void AnalyzeCompilation(CompilationStartAnalysisContext compilationStartContext, PXContext pxContext)
+        {
+            compilationStartContext.RegisterSyntaxNodeAction(syntaxContext => AnalyzeLocalizationMethodInvocation(syntaxContext, pxContext), SyntaxKind.InvocationExpression);
+        }
+
+        private void AnalyzeLocalizationMethodInvocation(SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            _invocationNode = syntaxContext.Node as InvocationExpressionSyntax;
+            if (_invocationNode == null || syntaxContext.CancellationToken.IsCancellationRequested)
+                return;
+
+            _syntaxContext = syntaxContext;
+            _pxContext = pxContext;
+
+            if (!IsLocalizationMethodInvocation())
+                return;
+
+            LocalizationMessageHelper messageHelper = new LocalizationMessageHelper(syntaxContext, pxContext, _messageExpression, _isFormatMethod);
+            messageHelper.ValidateMessage();
+        }
+
+        private bool IsLocalizationMethodInvocation()
+        {
+            if (Cancellation.IsCancellationRequested)
```

I'd suggest using `ThrowIfCancellationRequested()` instead of returning some definite value

---

Ok